libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -51,10 +57,21 @@ export class BookSearchComponent {
   }
 
   searchBooks() {
-    if (this.searchForm.value.term) {
-      this.store.dispatch(searchBooks({ term: this.searchTerm }));
-    } else {
-      this.store.dispatch(clearSearch());
-    }
+    // get search value from form
+    this.searchForm.valueChanges.pipe(
+        takeUntil(this.unsubscribe$),
+        debounceTime(500),
+        distinctUntilChanged()
+    ).subscribe(searchValue => {
+      return searchValue !== '' ?
+          this.store.dispatch(searchBooks({ term: this.searchTerm })) :
+          this.store.dispatch(clearSearch())
+    });
+  }
+
+  public ngOnDestroy() {
+    // unsubscribe to avoid data leaks
```

Comment:
You don't need to add a comment like this, it's a fairly common idiom.

Reply:
Removed comment.